src/types.ts

```diff
@@ -1,16 +1,160 @@
-import { Numbers } from "web3";
+import { Address, Numbers } from "web3";
 
-export interface NativeCurrencyData {
+export type NativeCurrencyData = {
   name: string;
   symbol: string;
   decimals: number;
-}
+};
 
-export interface AddEthereumChainRequest {
+/**
+ * Request to add a new chain to the user's wallet.
+ *
+ * See [EIP-3085](https://eips.ethereum.org/EIPS/eip-3085) for more details.
+ */
+export type AddEthereumChainRequest = {
   chainId: Numbers;
   blockExplorerUrls?: string[];
   chainName?: string;
   iconUrls?: string[];
   nativeCurrency?: NativeCurrencyData;
   rpcUrls?: string[];
-}
+};
+
+/**
+ * Request to switch to a new chain and register it with the user’s wallet if it isn’t already recognized.
+ *
+ * See [EIP-2015](https://eips.ethereum.org/EIPS/eip-2015) for more details.
+ */
+export type UpdateEthereumChainRequest = {
+  chainId: Numbers;
+  blockExplorerUrl?: string;
+  chainName?: string;
+  nativeCurrency?: NativeCurrencyData;
+  rpcUrls?: string[];
+};
+
+/**
+ * Request to switch the wallet’s currently active chain.
+ *
+ * See [EIP-3326](https://eips.ethereum.org/EIPS/eip-3326) for more details.
+ */
+export type SwitchEthereumChainRequest = {
+  chainId: Numbers;
+};
+
+/**
+ * Request to return a list of owned assets for the given address.
+ *
+ * See [EIP-2256](https://eips.ethereum.org/EIPS/eip-2256) for more details.
+ */
+export type GetOwnedAssetsRequest = {
+  /**
+   * Ethereum address that owns the assets.
+   */
+  address: Address;
+  options?: {
+    /**
+     * Chain ID respecting EIP-155.
+     */
+    chainId?: Numbers;
+    /**
+     * Maximum number of owned assets expected by the dApp to be returned.
+     */
+    limit?: number;
+    /**
+     * Array of asset interface identifiers such as ['ERC20', 'ERC721'].
+     */
+    types?: string[];
```

Comment:
For some of these less straightforward optional parameters, it may be nice to document what the behavior is when the parameter is omitted.